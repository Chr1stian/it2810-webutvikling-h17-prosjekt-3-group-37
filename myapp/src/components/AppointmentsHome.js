import React, { Component } from 'react';
import { CardText, CardTitle} from 'material-ui/Card';
import RaisedButton from 'material-ui/RaisedButton';

class AppointmentsHome extends Component {


  render(){

    return(
      <div>
        <CardTitle title="Appointment"></CardTitle>
        <CardText>Latest appointment</CardText>
        <RaisedButton label="Go to your appointments" style={{'display':'flex'}}/>
      </div>
    )


  }
}

export default AppointmentsHome;
