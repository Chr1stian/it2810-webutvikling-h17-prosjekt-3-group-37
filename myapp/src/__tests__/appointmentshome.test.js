import React from "react";
//Import Enzyme with shallow to test component as a unit
import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";
Enzyme.configure({ adapter: new Adapter() });
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
//Importing component that will be tested
import AppointmentsHome from "./../components/AppointmentsHome.js";
import "./../test/mock-localstorage.js";

//take snapshot of component
test("Component renders as expected", () => {
  const component = shallow(<AppointmentsHome />);
  let tree = toJson(component);
  expect(tree).toMatchSnapshot();
});

test("testing if short string are shortened or stays the same, should stay the same", () => {
  const component = shallow(<AppointmentsHome />);
  const string = "abcdefghijklmnopqrstuvwxyz";
  var newString = "";
  //formats new string if it is to long, if not it stays the same
  newString = component.instance().formatString(string);
  expect(newString).toBe(string);
})

test("testing if too long string are shortened", () => {
  const component = shallow(<AppointmentsHome />);
  const string = "abcdefghijklmnopqrstuvwxyz";
  var newString = "";
  //this makes string above 50 chars and therefore to long
  for (var i = 0; i < 2; i++) {
    newString = newString + string;
  }
  component.instance().formatString(newString);
  expect(newString).not.toBe(string);
})
