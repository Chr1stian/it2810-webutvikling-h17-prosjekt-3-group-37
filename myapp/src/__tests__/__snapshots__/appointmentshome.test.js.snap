// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders as expected 1`] = `
<div>
  <CardTitle
    style={
      Object {
        "paddingBottom": "0",
        "paddingTop": "4px",
        "textAlign": "center",
      }
    }
    title="Appointment"
  />
  <Divider
    inset={false}
  />
  <CardText>
    <div
      className="appointmentHomeInfoHeader"
    >
      <div>
        What
      </div>
      <div>
        When
      </div>
      <div>
        Where
      </div>
    </div>
    <Divider
      inset={false}
    />
    <div
      className="appointmentHomeInfo"
    >
      <div
        id="appointmentHomeTitle"
      >
        --
      </div>
      <div
        id="appointmentHomeTime"
      >
        <span>
          --
        </span>
        <span>
          --
        </span>
      </div>
      <div
        id="appointmentHomePlace"
      >
        --
      </div>
    </div>
  </CardText>
  <RaisedButton
    disabled={false}
    fullWidth={false}
    labelPosition="after"
    primary={false}
    secondary={true}
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Link
      className="navLink"
      replace={false}
      to="/appointments"
    >
      CHECK YOUR APPOINTMENTS
    </Link>
  </RaisedButton>
</div>
`;
